frontend/server/src/ZipToCdpConverter.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace OmegaUp;
+
+use ZipArchive;
+
+class ZipToCdpConverter{
+
+
+    public static function convert(string $zipFilePath, string $problemName): array {
+  
+        \OmegaUp\Validators::validateZipFileSize($zipFilePath);
+        \OmegaUp\Validators::validateZipIntegrity($zipFilePath);
+        \OmegaUp\Validators::validateStringNonEmpty($problemName, 'problemName');
+
+        // Procesar archivo
+        $zip = new ZipArchive();
+        $zip->open($zipFilePath);
+
+        // Construir CDP
+        $cdp = CdpBuilder::initialize($problemName);
+        self::populateBuilder($cdp, $zip);
+        CdpBuilder::buildCases($cdp);
+        CdpBuilder::processImages($cdp, $zip);
+
+        $zip->close();
+
+        return [
+            'status' => 'ok',
+            'message' => 'El ZIP fue convertido correctamente a CDP.',
+            'cdp' => CdpBuilder::build($cdp)
+        ];
+
+    }
+    private static function populateBuilder(array &$cdp, ZipArchive $zip): void {
+        $rgxStatement = '/^statements\/(en|es|pt)\.markdown$/';
+        $rgxSolutions = '/^solutions\/(en|es|pt)\.markdown$/';
+
+        ZipFileProcessor::iterateFiles(
+            $zip, 
+            function ($fileName) use (&$cdp, $rgxStatement, $rgxSolutions, $zip) {
```

Comment:
Aquí también hay que agregarle los tipos al parámetro del closure
```suggestion
            function (string $fileName) use (&$cdp, $rgxStatement, $rgxSolutions, $zip) {
```